src/lib/commandclass/ColorSwitchCC.ts

```diff
@@ -457,35 +480,39 @@ export class ColorSwitchCCReport extends ColorSwitchCC {
 		const valueId: ValueID = {
 			commandClass: CommandClasses["Color Switch"],
 			property: "currentColor",
-			endpoint: endpoint.index,
+			endpoint: this.endpointIndex,
 			propertyKey: colorName,
 		};
 
-		// TODO: Should we be setting this every run?
-		node.valueDB.setMetadata(valueId, {
-			...ValueMetadata.ReadOnlyNumber,
-			label: `Color ${colorName}`,
-			description: `The current level of the ${colorName} color.`,
-			min: 0,
-			max: 255,
-		});
-		node.valueDB.setValue(valueId, this._currentValue);
+		const valueDB = this.getValueDB();
+		if (!valueDB.hasMetadata(valueId)) {
+			valueDB.setMetadata(valueId, {
+				...ValueMetadata.ReadOnlyNumber,
+				label: `Color ${colorName}`,
+				description: `The current level of the ${colorName} color.`,
+				min: 0,
+				max: 255,
+			});
+		}
+		valueDB.setValue(valueId, this._currentValue);
 
 		if (this._targetValue != undefined) {
 			const targetValueId: ValueID = {
 				commandClass: CommandClasses["Color Switch"],
 				property: "targetColor",
-				endpoint: endpoint.index,
+				endpoint: this.endpointIndex,
 				propertyKey: colorName,
 			};
```

Comment:
This should probably be in a getXYZValueID method. I'm planning to rework the whole ValueID definition process and having them all at the top makes it easier.

Reply:
I replaced all valueID creation logic with functions.